src/ACESVGjsPanel.tsx

```diff
@@ -123,38 +132,41 @@ export class ACESVGPanel extends PureComponent<Props, PanelState> {
       }
       currentNode = svgWalker.nextNode() as Element;
     }
-    this.setState({ svgMappings: [...svgMappings], initialized: false });
-    this.props.options.svgMappings = [...svgMappings];
-    this.props.onOptionsChange(this.props.options);
-    this.forceUpdate();
+    this.props.onOptionsChange({
+      ...this.props.options,
+      svgMappings,
+    });
+    this.setState({ initialized: false });
   }
 
-  mappingClickHandler(event: React.MouseEvent<HTMLElement, MouseEvent>) {
+  mappingClickHandler = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
     if (event.target) {
       let clicked = event.target as Element;
-      let loopCount = 0;
-      let svgMappings: SVGIDMapping[] = [...this.state.svgMappings];
       if (clicked.id) {
+        let loopCount = 0;
         while (clicked.id === '') {
           loopCount++;
           if (loopCount > 20) {
+            console.log('no element found with an id');
             return;
           }
           clicked = clicked.parentNode as Element;
         }
-        for (let i = 0; i < svgMappings.length; i++) {
-          if (svgMappings[i].svgId === clicked.id) {
-            return;
-          }
+
+        if (this.props.options.svgMappings.find(m => m.svgId === clicked.id)) {
+          console.log('mapping already exists', clicked);
+          return;
         }
-        svgMappings.push({ svgId: clicked.id, mappedName: '' });
-        this.setState({ svgMappings: [...svgMappings], initialized: false });
-        this.props.options.svgMappings = [...svgMappings];
-        this.props.onOptionsChange(this.props.options);
-        this.forceUpdate();
+
+        this.props.onOptionsChange({
+          ...this.props.options,
+          svgMappings: [...this.props.options.svgMappings, { svgId: clicked.id, mappedName: '' }],
```

Comment:
again, avoiding the direct mutating... 

but sadly, the render still happens on the next activity

Reply:
I mean look at the commit log, I've had that exact syntax in there at least 4 different times, it just doesn't work.

Reply:
Specifically the render happens on the next activity of the editor component, not the next activity of the panel component. It seems there needs to be some sort of event we can bubble up to the parent of both the panel and the editor to let them know they need to re-render, it looks like it's just editor->panel instead of editor<->panel